docs/proposals/20240807-in-place-updates.md

```diff
@@ -285,28 +291,28 @@ box Management Cluster
     participant hook as External updater
 end
 
-Operator->>apiserver: make changes to MD <br>(ExternalUpdate strategy)
+Operator->>apiserver: make changes to MD
 apiserver->>capi: Notify changes
 apiserver->>Operator: OK
-capi->>capi: Define Update Plan
+capi->>capi: Decide Update Strategy
 capi->>apiserver: Create new MachineSet
 loop For all machines
-    capi->>apiserver: Add Update Plan to Machine<br> and move to new Machine Set
+    capi->>apiserver: Mark as pending and move to new Machine Set
     apiserver->>msc: Notify changes
-    msc->>apiserver: Update Machine Spec
+    msc->>apiserver: Update Machine Spec and set UpToDate condition
     apiserver->>mach: Notify changes
     loop For all updaters in plan
         mach->>hook: Run updater
     end
-    mach->>apiserver: Mark Machine as updated
+    mach->>apiserver: Mark Machine as done
+    msc->>apiserver: Set UpToDate condition
 end
 ```
 
-The MachineDeployment controller updates machines in place in a very similar way to rolling updates: by creating a new MachineSet and moving the machines from the old MS to the new one. We want to stress that the Machine objects won't be deleted and recreated like in the current rolling strategy. The MachineDeployment will just update the OwnerRefs, effectively moving the existing Machine object from one MS to another. The number of machines moved at once might be made configurable on the MachineDeployment in the same way `maxSurge` and `maxUnavailable` control this for rolling updates.
-
-When the new MachineSet controller sees a new Machine with an outdated spec, it updates the spec to match the one in the MS. This update is what triggers the Machine controller to start executing the external updaters.
+The MachineDeployment controller updates machines in place in a very similar way to rolling updates: by creating a new MachineSet and moving the machines from the old MS to the new one. We want to stress that the Machine objects won't be deleted and recreated like in the current rolling strategy. The MachineDeployment will just update the OwnerRefs and labels, effectively moving the existing Machine object from one MS to another. The number of machines moved at once might be made configurable on the MachineDeployment in the same way `maxSurge` and `maxUnavailable` control this for rolling updates.
 
-> TODO: we might want the MachineDeployment to store the update plan in the MachineSet and have the MachineSet pass it down to the Machine. This should not change the high level design, so the decision will come after this first draft.
+When the new MachineSet controller sees a new Machine with an outdated spec, it updates the spec to match the one in the MS. This update together with marking machine as pending and setting a condition is what triggers the Machine controller to start executing 
```

Comment:
who sets the hook annotation and the uptodate condition here? MD controller or MS controller?